presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -313,6 +315,34 @@ protected Boolean visitInPredicate(InPredicate node, Void context)
         protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
             if (metadata.isAggregationFunction(node.getName())) {
+                if (node.getOrderBy().isPresent()) {
+                    List<Expression> sortKeys = node.getOrderBy().get().getSortItems().stream()
+                            .map(SortItem::getSortKey)
+                            .collect(toImmutableList());
+                    if (node.isDistinct()) {
+                        List<FieldId> argumentReferences = node.getArguments().stream()
+                                .map(NodeRef::of)
+                                .map(columnReferences::get)
+                                .collect(toImmutableList());
+                        for (Expression sortKey : sortKeys) {
+                            if (!argumentReferences.contains(columnReferences.get(NodeRef.of(sortKey)))) {
```

Comment:
I think I handled complex expression already. Could you remind me with an example what you think might not work? Thanks!